superset-frontend/src/views/CRUD/utils.tsx

```diff
@@ -380,3 +393,57 @@ export const hasTerminalValidation = (errors: Record<string, any>[]) =>
             isNeedsPassword(payload) || isAlreadyExists(payload),
         ),
   );
+
+function buildDashboardTabsTree(
+  nodeId: string,
+  data: { [key: string]: any },
+  tree: any,
+  traversedIds: any,
+) {
+  if (nodeId in traversedIds) {
+    return traversedIds[nodeId];
+  }
+  const element = data[nodeId];
+
+  const parents: string[] = element.parents.filter(
+    (parent: string) => data[parent] && data[parent].type === 'TAB',
+  );
+  const newNode = {
+    key: nodeId,
+    title: element.meta.text,
+    children: [],
+  };
+
+  // eslint-disable-next-line no-param-reassign
+  traversedIds[nodeId] = newNode;
+  if (parents.length === 0) {
+    tree.push(newNode);
+
+    return newNode;
+  }
+  // Only take the last parent.
+  const parent = parents[parents.length - 1];
+  const parentNode = buildDashboardTabsTree(parent, data, tree, traversedIds);
+  parentNode.children.push(newNode);
+  return newNode;
+}
+
+export function useDashboardTabTree(id: string | number) {
+  const tree: DataNode[] = [];
+  const traversed = {};
+  const dashboardInfo = useDashboard(id);
+  const positionData: PositionData | null = dashboardInfo.result?.position_data;
+
+  if (!positionData) {
+    return null;
+  }
+  Object.entries(positionData).forEach(([_, component]) => {
+    if (['TABS'].includes(component.type)) {
+      // @ts-ignore
```

Comment:
what are we ts-ignoring? can we avoid it, or instead use ts-expect-error with a description of what the error is?